email_notification/helpers/email_sender.py

```diff
@@ -94,19 +90,46 @@ def send_application_email(self, *, application: Application, forced_language: L
 
         self._send_email(message_builder)
 
+    def send_batch_application_emails(self, *, applications: Iterable[Application]):
+        # Sort recipients by language, so that we can batch send the emails in the correct language
+        # and avoid sending the same email to the same recipient multiple times.
+        all_recipients: set[str] = set()
+        recipients_by_language: dict[LanguageType, set[str]] = {"fi": set(), "sv": set(), "en": set()}
+        for application in applications:
+            user_language = safe_getattr(application.user, "preferred_language") or settings.LANGUAGE_CODE
```

Comment:
Could simply use `getattr` in this function since not fetching nested attributes?

Reply:
Yes we could, but then we would also require to give it a `default=None` argument, which is now handled by `safe_getattr ` and this seemed more readable to me :)